components/script/dom/webglshader.rs

```diff
@@ -114,15 +117,28 @@ impl WebGLShader {
                     // It could be interesting to retrieve the info log from the paint thread though
                     let msg = WebGLCommand::CompileShader(self.id, translated_source);
                     self.renderer.send(CanvasMsg::WebGL(msg)).unwrap();
-                    self.compilation_status.set(ShaderCompilationStatus::Succeeded);
+
+                    match self.parameter(constants::COMPILE_STATUS) {
```

Comment:
We should be pretty confident in that if the shader validation succeeds, the compilation is also successful for the context we're targeting. Do you have an example of something that fails in this case? If that's the case, that's either an ANGLE bug, or that we're using the wrong kind of GL context.